cmd/apichecker/apicheck.go

```diff
@@ -0,0 +1,1418 @@
+	flag.StringVar(&trelloChecklistID, "checklistid", "", "sets the checklist ID for Trello board interaction")
+	flag.StringVar(&trelloCardID, "cardid", "", "sets the card ID for Trello board interaction")
+	flag.StringVar(&trelloListID, "listid", "", "sets the list ID for Trello board interaction")
+	flag.StringVar(&trelloBoardID, "boardid", "", "sets the board ID for Trello board interaction")
+	flag.StringVar(&trelloUsername, "trellousername", "", "sets the username for Trello board interaction")
+	flag.StringVar(&exchangeName, "exchangename", "", "sets the exchangeName for the new exchange")
+	flag.StringVar(&checkType, "checktype", "", "sets the checkType for the new exchange")
+	flag.StringVar(&tokenData, "tokendata", "", "sets the tokenData for adding a new exchange")
+	flag.StringVar(&key, "key", "", "sets the key for adding a new exchange")
+	flag.StringVar(&val, "val", "", "sets the val for adding a new exchange")
+	flag.StringVar(&tokenDataEnd, "tokendataend", "", "sets the tokenDataEnd for adding a new exchange")
+	flag.StringVar(&textTokenData, "texttokendata", "", "sets the textTokenData for adding a new exchange")
+	flag.StringVar(&regExp, "regexp", "", "sets the regExp for adding a new exchange")
+	flag.StringVar(&dateFormat, "dateformat", "", "sets the dateFormat for adding a new exchange")
+	flag.StringVar(&path, "path", "", "sets the path for adding a new exchange")
+	flag.BoolVar(&add, "add", false, "used as a trigger to add a new exchange from command line")
+	flag.BoolVar(&verbose, "verbose", false, "increases logging verbosity for API Update Checker")
+	flag.Parse()
+	var err error
+	configData, err = ReadFileData(jsonFile)
+	if err != nil {
+		log.Fatal(err)
+	}
+	testConfigData, err = ReadFileData(testJSONFile)
+	if err != nil {
+		log.Fatal(err)
+	}
+	usageData = testConfigData
+	if CanUpdateTrello() {
+		usageData = configData
+	}
+	if add {
+		switch checkType {
+		case github:
+			var data GithubData
+			data.Repo = path
+			err = Add(exchangeName, checkType, data, false)
+			if err != nil {
+				log.Fatal(err)
+			}
+		case htmlScrape:
+			var data HTMLScrapingData
+			data.TokenData = tokenData
+			data.Key = key
+			data.Val = val
+			data.TokenDataEnd = tokenDataEnd
+			data.TextTokenData = textTokenData
+			data.DateFormat = dateFormat
+			data.RegExp = regExp
+			data.Path = path
+			err = Add(exchangeName, checkType, data, false)
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+	}
+	if CanUpdateTrello() {
+		SetAuthVars()
+		err = UpdateFile(backupFile)
+		if err != nil {
+			log.Fatal(err)
+		}
+		err = CheckUpdates(jsonFile)
+		if err != nil {
+			log.Fatal(err)
+		}
+	} else {
+		log.Printf("This is a test update since API keys are not set.\n")
+		err := CheckUpdates(testJSONFile)
+		if err != nil {
+			log.Fatal(err)
+		}
+		log.Printf("API update check completed successfully")
+	}
+}
+
+// SetAuthVars checks if the cmdline vars are set and sets them onto config file and vice versa
+func SetAuthVars() {
```

Comment:
When creating standalone apps, there's no need to export func names (start with a capital letter), unless you plan to use these in others modules, but we're not in this context